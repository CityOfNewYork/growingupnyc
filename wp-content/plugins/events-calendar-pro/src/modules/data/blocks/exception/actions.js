/**
 * External dependencies
 */
import { curry } from 'lodash/fp';

/**
 * Internal dependencies
 */
import * as types from './types';

export const addField = () => ( {
	type: types.ADD_EXCEPTION_FIELD,
} );

export const addException = ( payload ) => ( {
	type: types.ADD_EXCEPTION,
	payload,
} );

export const removeField = () => ( {
	type: types.REMOVE_EXCEPTION_FIELD,
} );

export const removeException = ( index ) => ( {
	type: types.REMOVE_EXCEPTION,
	index,
} );

export const editException = curry( ( index, payload ) => ( {
	type: types.EDIT_EXCEPTION,
	index,
	payload,
} ) );

export const syncException = curry( ( index, payload ) => ( {
	type: types.EDIT_EXCEPTION,
	index,
	payload,
	sync: true,
} ) );

export const syncExceptionsFromDB = payload => ( {
	type: types.SYNC_EXCEPTIONS_FROM_DB,
	payload,
} );
