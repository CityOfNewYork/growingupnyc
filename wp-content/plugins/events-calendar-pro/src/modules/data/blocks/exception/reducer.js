/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import * as types from './types';

function edit( state, action ) {
	const field = Object.assign( {}, state[ action.index ], action.payload );

	if ( state.length === 1 ) {
		return [ field ];
	}
	return [
		...state.slice( 0, action.index ),
		field,
		...state.slice( action.index + 1 ),
	];
}

export default function exception( state = [], action ) {
	switch ( action.type ) {
		case types.ADD_EXCEPTION:
			return [
				...state,
				action.payload,
			];
		case types.EDIT_EXCEPTION:
			return edit( state, action );
		case types.REMOVE_EXCEPTION:
			return state.filter( ( _, index ) => index !== action.index );
		case types.SYNC_EXCEPTIONS_FROM_DB:
			return JSON.parse( action.payload );
		default:
			return state;
	}
}
