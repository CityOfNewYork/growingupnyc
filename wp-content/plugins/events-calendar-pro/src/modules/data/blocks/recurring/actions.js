/**
 * External dependencies
 */
import { curry } from 'lodash/fp';

/**
 * Internal dependencies
 */
import * as types from './types';

export const addField = () => ( {
	type: types.ADD_RULE_FIELD,
} );

export const addRule = ( payload ) => ( {
	type: types.ADD_RULE,
	payload,
} );

export const removeField = () => ( {
	type: types.REMOVE_RULE_FIELD,
} );

export const removeRule = ( index ) => ( {
	type: types.REMOVE_RULE,
	index,
} );

export const editRule = curry( ( index, payload ) => ( {
	type: types.EDIT_RULE,
	index,
	payload,
} ) );

export const syncRule = curry( ( index, payload ) => ( {
	...editRule( index, payload ),
	sync: true,
} ) );

export const syncRulesFromDB = ( payload ) => ( {
	type: types.SYNC_RULES_FROM_DB,
	payload,
} );
