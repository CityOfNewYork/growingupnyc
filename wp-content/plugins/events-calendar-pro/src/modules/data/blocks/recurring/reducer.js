/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import * as types from './types';

function edit( state, action ) {
	const field = Object.assign( {}, state[ action.index ], action.payload );

	if ( state.length === 1 ) {
		return [ field ];
	}
	return [
		...state.slice( 0, action.index ),
		field,
		...state.slice( action.index + 1 ),
	];
}

export default function recurring( state = [], action ) {
	switch ( action.type ) {
		case types.ADD_RULE:
			return [
				...state,
				action.payload,
			];
		case types.EDIT_RULE:
			return edit( state, action );
		case types.REMOVE_RULE:
			return state.filter( ( _, index ) => index !== action.index );
		case types.SYNC_RULES_FROM_DB:
			return JSON.parse( action.payload );
		default:
			return state;
	}
}
