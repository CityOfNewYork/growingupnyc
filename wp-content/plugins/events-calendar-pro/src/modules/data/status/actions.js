/**
 * Internal dependencies
 */
import * as types from './types';

export const setSeriesQueueStatus = ( payload ) => ( {
	type: types.SET_SERIES_QUEUE_STATUS,
	payload,
} );
