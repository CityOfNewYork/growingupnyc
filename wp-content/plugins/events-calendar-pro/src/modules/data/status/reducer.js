/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_SERIES_QUEUE_STATUS:
			return {
				...state,
				...action.payload,
			};
		default:
			return state;
	}
};
