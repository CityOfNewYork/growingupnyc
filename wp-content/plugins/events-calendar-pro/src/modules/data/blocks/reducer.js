/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import recurring from '@moderntribe/events-pro/data/blocks/recurring/reducer';
import exception from '@moderntribe/events-pro/data/blocks/exception/reducer';
import additionalFields from '@moderntribe/events-pro/data/blocks/additional-fields';

export default combineReducers( {
	recurring,
	exception,
	additionalFields,
} );
