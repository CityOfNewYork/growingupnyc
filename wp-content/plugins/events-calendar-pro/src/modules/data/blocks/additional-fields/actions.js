/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { SET_ADDITIONAL_FIELD_OUTPUT } from './types';

export const setInitialState = ( props ) => ( {
	type: types.SET_ADDITIONAL_FIELD_INITIAL_STATE,
	payload: props,
} );

export const setFieldChange = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_CHANGE,
	payload: {
		name,
	},
} );

export const setFieldBlur = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_BLUR,
	payload: {
		name,
	},
} );

export const setFieldBlurWithType = ( name, type ) => ( {
	type: `${ types.SET_ADDITIONAL_FIELD_BLUR }/${ type }`,
	payload: {
		name,
	},
} );

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const appendFieldValue = ( name, value ) => ( {
	type: types.APPEND_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const removeFieldValue = ( name, value ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

export const setFieldMetaKey = ( name, metaKey ) => ( {
	type: types.SET_ADDITIONAL_FIELD_META_KEY,
	payload: {
		name,
		metaKey,
	},
} );

export const setFieldOutput = ( name, output ) => ( {
	type: SET_ADDITIONAL_FIELD_OUTPUT,
	payload: {
		name,
		output,
	},
} );
