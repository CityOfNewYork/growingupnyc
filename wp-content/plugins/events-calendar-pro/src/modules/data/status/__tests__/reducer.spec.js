/**
 * Internal dependencies
 */
import * as actions from '../actions';
import reducer, { DEFAULT_STATE } from '../reducer';

describe( '[STORE] - Status reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should set status', () => {
		const current = reducer( DEFAULT_STATE, actions.setSeriesQueueStatus( { cool: true } ) );
		expect( current ).toMatchSnapshot();
	} );
} );
