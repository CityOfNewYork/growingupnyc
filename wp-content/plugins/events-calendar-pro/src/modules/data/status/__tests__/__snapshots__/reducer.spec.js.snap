// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Status reducer Should set status 1`] = `
Object {
  "cool": true,
}
`;
